### PR TITLE
fix: simple transaction do not have any 4 inputs limitation

### PR DESCRIPTION
**Pull request by Alex-Werner**
## Issue being fixed or feature implemented

As reported in #157, the SimpleTransactionOptimizedAccumulator strategy was optimising for a maximum of 4 inputs to ensure they get locked. By default, this was the transaction strategy that were being used, so anyone that will try to spend using default options would encounter that case (if using small inputs), and the documentation would not explained the why.
As this rules actually is not in place anymore, we can safely modify the strategy to not put this limitation anymore.

## What was done?
- Remove 4 inputs limitation for the defaut accumulator used.
- Switched to simpleDescendingStrategy
- Improve documentation on strategy usage.

## How Has This Been Tested?
Rely on differents UTXO selection test we have + additional test in coinSelection file (was already tested in Account.createTransaction)

## Breaking Changes
N/A (but won't cause the issue again)

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/CONSTANTS.js [hunk 1]
```diff
@@ -27,8 +27,8 @@ const CONSTANTS = {
   UTXO_SELECTION_MAX_FEE_VS_SINGLE_UTXO_FEE_FACTOR: 5,
   MAX_STANDARD_TX_SIZE: 100000,
   MAX_P2SH_SIGOPS: 15,
-  MAX_INPUTS_FOR_AUTO_IX: 4,
-  UTXO_MAX_INPUTS_PER_TX: 25,
+  // limit to how many time an unconfirmed input in a new tx can be respend
```
Comment (Cofresi) [inline src/CONSTANTS.js:30]:
```suggestion
  // limit to how many times an unconfirmed input in a new tx can be respent
```

File: src/utils/coinSelection.spec.js [hunk 2]
```diff
@@ -317,6 +317,51 @@ describe('Utils - coinSelection', function suite() {
...
+
+      txEstimator.addOutputs(outputsList);
+
+      let inputValue = 0;
+      let outputValue = txEstimator.getOutValue();
+      const randomlySelectedUtxos = [];
+
+      while(inputValue<outputValue){
+        if(copiedUtxos.length === 0){
+          throw new Error('Not enought utxo');
```
Comment (thephez) [inline src/utils/coinSelection.spec.js:334]:
```suggestion
          throw new Error('Not enought UTXOs');
```

File: src/utils/coinSelection.spec.js [hunk 3]
```diff
@@ -331,7 +331,7 @@ describe('Utils - coinSelection', function suite() {
 
       while(inputValue<outputValue){
         if(copiedUtxos.length === 0){
-          throw new Error('Not enought utxo');
+          throw new Error('Not enought UTXOs');
```
Comment (thephez) [inline src/utils/coinSelection.spec.js:334]: Oops, missed this the first time :see_no_evil: 
```suggestion
          throw new Error('Not enough UTXOs');
```

Comment (Alex-Werner) [reply src/utils/coinSelection.spec.js:334]: It's already merged, but I do will update that on another PR, I've taken note of it on my todo :) Thank you :)

**Pull request merged**